File: SchedulingAPI/UserManagement/Models/Role.cs
using CodebookManagement.Attributes;
using CodebookManagement.Models;
using SQLContext.Attributes;
using SQLContext.Models;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using Web.Attributes;
using ColumnAttribute = CodebookManagement.Attributes.ColumnAttribute;

namespace UserManagement.Models
{
    public class Role : ICodebook
    {
        [PrimaryKey]
        public int RoleId { get; set; }
        [Column(ControlType.Input, true)]
        public string Code { get; set; }
        [Column(ControlType.Input, true)]
        public string Name { get; set; }
        [Column(ControlType.Toggle, true)]
        public bool Active { get; set; }
        public int UserId { get; set; }

        public int Id { get; set; }

        public IEnumerable<Menu> Menus { get; set; }
        public IEnumerable<Permission> Permissions { get; set; }

        [Join(JoinType.Inner, "CompanyId", "CompanyId")]
        [ChildValidation()]
        public User User { get; set; } = new User();

        public Role() { }
    }
}

File: SchedulingFront-OLD/src/app/common/models/role.ts
import { Menu } from './menu';
import { Permission } from './permission';

export class Role {

    roleId: number;
    code: string;
    name: string;
    active: boolean;
    menus: Array<Menu>;
    permissions: Array<Permission>;

    constructor() {
        this.code = null;
        this.roleId = null;
        this.name = null;
        this.active = true;
        this.menus = [];
        this.permissions = [];
    }
}
